src/compiler.rs

```diff
@@ -1492,6 +1492,41 @@ impl Compiler {
         }
     }
 
+    fn for_loop_condition(&mut self, block: &mut Block) {
+        expect!(
+            self,
+            Token::Comma,
+            "Expect ',' between initalizer and loop expression"
+        );
+
+        let cond = block.curr();
+        self.expression(block);
+        let cond_out = add_op(self, block, Op::Illegal);
+        let cond_cont = add_op(self, block, Op::Illegal);
+        expect!(
+            self,
+            Token::Comma,
+            "Expect ',' between initalizer and loop expression"
+        );
+
+        let inc = block.curr();
+        push_scope!(self, block, {
+            self.statement(block);
+        });
+        add_op(self, block, Op::Jmp(cond));
+
+        // patch_jmp!(Op::Jmp, cond_cont => block.curr());
+        block.patch(Op::Jmp(block.curr()), cond_cont);
+        self.scope(block);
+        add_op(self, block, Op::Jmp(inc));
```

Comment:
We should figure out if we want to do something like this macro or not. I still think that `block.patch()` is a bit hard to follow but maybe the alternative (something that looks more like `push_scope!`) is even harder to read.

Reply:
I at least think we should switch the parameter ordering so it's read as "`patch` `cond_cont` to a `Op::Jmp(block.curr())`".